src/components/header.jsx

```diff
@@ -111,48 +107,42 @@ const NavLink = styled.a`
   }
 `;
 
-const Header = (props) => {
-  const { defaultSelected, router } = props;
+const Header = ({ defaultSelected, router }) => {
   const classes = useStyles();
+  const [session, loading] = useSession();
+
   const [anchorEl, setAnchorEl] = React.useState(null);
   const [selected, setSelected] = React.useState(defaultSelected);
   const [filteredRoutes, setFilteredRoutes] = React.useState([]);
-  const [session, loading] = useSession();
 
-  // console.log(currentUser);
-  // let filteredRoutes = [];
-  // const [filteredRoutes, setFilteredRoutes] = React.useState([]);
+  React.useEffect(() => async () => {
+    if (!loading && session && filteredRoutes.length == 0) {
+      const json = await queryUser();
+      setFilteredRoutes(filterRoutes(json.payload));
+    }
+  });
 
   const filterRoutes = (currentUser) => {
-    let fRoutes = [];
-    if (currentUser.type == "Admin") {
-      fRoutes = routes.filter(
-        (item) => item.type == "Admin" || item.type == "All"
-      );
-    } else if (currentUser.type == "BusDriver") {
-      fRoutes = routes.filter(
-        (item) => item.type == "BusDriver" || item.type == "All"
-      );
-    } else if (currentUser.type == "ClubDirector") {
-      fRoutes = routes.filter(
-        (item) =>
-          item.type == "ClubDirectorAttendanceClerk" || item.type == "All"
-      );
-    } else if (currentUser.type == "AttendanceClerk") {
-      fRoutes = routes.filter(
-        (item) =>
-          item.type == "ClubDirectorAttendanceClerk" || item.type == "All"
-      );
-    } else {
-      fRoutes = routes.filter((item) => item.type == "All");
+    let allRoutes = routes.filter(item => item.type === "All");
+
+    let userRoutes = [];
+    if (currentUser.type === "Admin") {
```

Comment:
We should consider extracting the user types into an enum. Currently of the type checking is hardcoded to check the string value and duplicated all throughout the codebase.

Reply:
I think I've left previous comments similar to this on Katsuki's PRs. Work with him on removing the magic strings